packages/components/src/internal/components/labels/PrintLabelsModal.tsx

```diff
@@ -63,10 +66,13 @@ export class PrintLabelsModalImpl extends PureComponent<Props & InjectedQueryMod
     }
 
     componentDidMount() {
+        const { model, schemaName, queryName } = this.props;
+
         this.props.actions.addModel(
             {
                 id: this._modelId,
-                schemaQuery: SchemaQuery.create(this.props.schemaName, this.props.queryName),
+                schemaQuery: model?.schemaQuery ?? SchemaQuery.create(schemaName, queryName),
+                baseFilters: model?.filters,
```

Comment:
Looks like this is creating a new version of the model if it is provided. Is there a need for that?

Reply:
I think so, we are manipulating the selections and some of the model metadata/display properties which I don't think we want propagated to the initiating model. I think we could refactor some of that to state properties or callback/effects but seems like a bit more of a refactor than intended for this.

Reply:
Agreed. Thanks.